x-pack/filebeat/input/azureblobstorage/auth_test.go

```diff
@@ -0,0 +1,176 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package azureblobstorage
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/h2non/gock"
```

Comment:
Do we need this? From what I can see, an implementation of `http.RoundTripper` would satisfy the testing requirements that tests here add, arguably more cleanly.

Reply:
I had the same thought when I gave this a quick scan yesterday. The declarative setup is nice, but I'm not sure it's worth another dependency.

Reply:
If we go that route then we would lose the context to the serv.URL thats local to each test. serv.URL would be required to set & override the urls in the response JSON without which the sdk will always try to call 
```https://login.microsoftonline.com``` for the token validation check, as this is hardwired into the sdk and cannot be overridden by conventional means.

Reply:
> If we go that route then we would lose the context to the serv.URL thats local to each test.

Why? There is nothing in the gock code that could not be done easily and more transparently with local code. AFAICS the mock code provides author convenience at the cost of transparency for the maintainer.

Reply:
@efd6, have removed gock and simplified the test.